spyder/app/mainwindow.py

```diff
@@ -1396,6 +1397,10 @@ def post_visible_setup(self):
         if not running_under_pytest():
             self.report_missing_dependencies()
 
+        # Show tour dialog
+        self.tour_dialog = tour.OpenTourDialog(self, lambda: self.show_tour(0))
+        self.tour_dialog.show()

```

Comment:
I think this makes the dialog to be displayed every time Spyder starts. To avoid that, you could connect a method called `show_tour_message` to the signal called `sig_setup_finished` (emitted at the end of `post_visibile_setup`), at the end of the `__init__` method of `MainWindow`:

```python
...

# To show the message about starting the tour
sig_setup_finished.connect(self.show_tour_message)

# Apply preferences
self.apply_settings()

# To set all dockwidgets tabs to be on top (in case we want to do it
...
```

Then you'd need to declare that method like this

```python
@Slot()
def show_tour_message(self):
    """Show message about starting the tour the first time Spyder starts."""
    show_tour = CONF.get('main', 'show_tour_message')
    if show_tour:
        CONF.set('main', 'show_tour_message', False)
        self.tour_dialog = tour.OpenTourDialog(self, lambda: self.show_tour(0))
        self.tour_dialog.show()
```

And finally you'd need to set the `show_tour_message` option as `True` in the `main` section of our defaults preferences, which are saved in `spyder/config/main.py`

Reply:
IMO, we should also not show the tour dialog when Spyder is launched in dev mode (to save us from having to click through it every time we launch Spyder in safe mode to test bugs and PRs, as we do dozens of times per day), i.e. when `DEV` is truthy, and also when `PYTEST` is truthy unless we want to write a test for it (which we really should). @ccordoba12 thoughts?

Reply:
> IMO, we should also not show the tour dialog when Spyder is launched in dev mode (to save us from having to click through it every time we launch Spyder in safe mode to test bugs and PRs

Right, I agree with that. @juanis2112, please also avoid showing the dialog when `DEV` is `True`. That constant is present in `spyder/config/base.py`and it's `True` when we're running Spyder from `bootstrap.py`.

Reply:
Actually, this will prevent us from actually testing it, manually or otherise. I was in the process of submitting a revised suggestion that instead checks for the safe mode env variable, so that if we're intentionally running Spyder under safe mode then it will not nag us (or the user) with the popup, while simply running Spyder under the normal dev environment with settings reset will, which we need to do anyway to properly test the only-run-once behavior. Standby.